user-impl/src/main/java/com/example/auction/user/impl/UserServiceImpl.java

```diff
@@ -42,10 +42,16 @@ public UserServiceImpl(PersistentEntityRegistry registry, ActorSystem system, Ma
         return user -> {
             UUID uuid = UUID.randomUUID();
             String password = PUserCommand.hashPassword(user.getPassword());
-            PUser pUser = new PUser(uuid, user.getName(), user.getEmail(),password);
-            return entityRef(uuid)
-                    .ask(new PUserCommand.CreatePUser(pUser))
-                    .thenApply(done -> Mappers.toApi(pUser));
+            entityRef(uuid)
+                    .ask(new PUserCommand.CreatePUser(user.getName(), user.getEmail(),password));
+
+            return entityRef(uuid).ask(PUserCommand.GetPUser.INSTANCE).thenApply(maybeUser -> {
```

Comment:
Sorry to push this back again, but I didn't mean to call `GetPUser`. The `CreatePUser` command returns the newly created user itself:

```java
return entityRef(uuid)
  .ask(new PUserCommand.CreatePUser(user.getName(), user.getEmail(), password))
  .thenApply(createdUser -> Mappers.toApi(createdUser));
```

The reason this is important is that each `ask` call is potentially a trip over the network to request the information from the entity. Lagom is designed for high-performance applications, so we want our example code to demonstrate best practices, which include minimizing unnecessary network traffic and other redundant work.

Furthermore, the code is simpler with just the one command. There is no need to handle the case of the user not being present, because we know that if it got this far, the user creation was successful. As you said before, success lies in doing less! 😄